apps/system/lockscreen/js/charging.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+(function(exports) {
+
+/**
+ * Creates an object used for refreshing the charging string UI element.
+ * @class  ChargingStatus
+ */
+function ChargingStatus() {
+  this.refreshCB = null;
+}
+
+ChargingStatus.prototype.start = function cs_start() {
+  this.refresh();
+
+  if (!this.refreshCB) {
+    this.refreshCB = this.refresh.bind(this);
+
+    navigator.battery.addEventListener('chargingchange',
+      this.refreshCB);
+    navigator.battery.addEventListener('levelchange',
+      this.refreshCB);
+    navigator.battery.addEventListener('chargingtimechange',
+      this.refreshCB);
+  }
+};
+
+ChargingStatus.prototype.stop = function cs_stop() {
+  if (this.refreshCB) {
```

Comment:
If you use the DOMEvent pattern I commented this method could be:

  this.listenEvents.forEach((ename) => {
    navigator.battery.removeEventListener(ename);
  });